habitat-lab/habitat/tasks/rearrange/sub_tasks/nav_to_obj_sensors.py

```diff
@@ -208,7 +273,17 @@ def reset_metric(self, *args, **kwargs):
         )
 
     def update_metric(self, *args, episode, task, observations, **kwargs):
-        targ = task.nav_goal_pos
+        if len(task.nav_goal_pos.shape) == 2:
+            # RotDist to closest goal
+            closest_goal = np.argmin(
```

Comment:
Same geodesic_distance calculation issue here

Reply:
Here, we are computing base's angle to the closest goal. We need the individual geodesic distances, so I am letting the loop over goals stay.

Reply:
Can you benchmark how long this takes? If its too long we can ask if these functions can be added on the cpp side?

Reply:
Will do.

Reply:
I replaced this loop with a single multi-goal geodesic call and didn't see a noticeable difference in training times. This may be because we have too few goals here (1-2 for most episodes, up to 5-7 in a few rare cases).

Reply:
Replaced argmin with a single call to MultiGoalShortestPath API.